Tests/Common/Communication/EventStorageTests.swift

```diff
@@ -0,0 +1,165 @@
+
+    func test_storeEvent_givenValidEvent_expectFileExists() async throws {
+        let event = ProfileIdentifiedEvent(identifier: "testID")
+        try await eventStorageManager.store(event: event)
+
+        let eventFileURL = await eventStorageManager.baseDirectory
+            .appendingPathComponent(ProfileIdentifiedEvent.key)
+            .appendingPathComponent("\(event.storageId).json")
+        let fileExists = FileManager.default.fileExists(atPath: eventFileURL.path)
+        XCTAssertTrue(fileExists, "Event file should exist after storing")
+    }
+
+    func test_storeEvent_givenNewEventType_expectDirectoryCreated() async throws {
+        let event = ResetEvent() // Assume this is a new event type
+        try await eventStorageManager.store(event: event)
+
+        let eventTypeDirectory = await eventStorageManager.baseDirectory.appendingPathComponent(ResetEvent.key)
+        let directoryExists = FileManager.default.fileExists(atPath: eventTypeDirectory.path)
+        XCTAssertTrue(directoryExists, "Event storage should create the directory if it does not exist")
+    }
+
+    // MARK: - Event Retrieval Tests
+
+    func test_loadEvents_givenStoredEvent_expectEventLoaded() async throws {
+        let event = ProfileIdentifiedEvent(identifier: "testID")
+        try await eventStorageManager.store(event: event)
+
+        let loadedEvents = try await eventStorageManager.loadEvents(ofType: ProfileIdentifiedEvent.key)
+        let containsEvent = loadedEvents.contains { ($0 as? ProfileIdentifiedEvent)?.identifier == "testID" }
+        XCTAssertTrue(containsEvent, "Loaded events should contain the stored event")
+    }
+
+    func test_loadEvents_givenNonExistentDirectory_expectEmptyArray() async throws {
+        let nonExistentEventType = "NonExistentEvent"
+        let loadedEvents = try await eventStorageManager.loadEvents(ofType: nonExistentEventType)
+        XCTAssertTrue(loadedEvents.isEmpty, "Loading events for a non-existent directory should return an empty array")
+    }
+
+    func test_loadEvents_givenNonexistentEventType_expectEmptyArray() async throws {
+        let loadedEvents = try await eventStorageManager.loadEvents(ofType: "NonexistentType")
+        XCTAssertTrue(loadedEvents.isEmpty, "Loading events for a nonexistent type should return an empty array")
+    }
+
+    func test_loadEvents_givenCorruptDataInFile_expectValidEventsLoaded() async throws {
+        // Ensure the directory for the event type exists
+        let eventTypeDirectory = await eventStorageManager.baseDirectory.appendingPathComponent(ProfileIdentifiedEvent.key)
+        try FileManager.default.createDirectory(at: eventTypeDirectory, withIntermediateDirectories: true)
+
+        // Store a corrupt event file manually
+        let corruptData = Data("corrupt data".utf8)
+        let corruptFileURL = eventTypeDirectory.appendingPathComponent("corruptEvent.json")
+        try corruptData.write(to: corruptFileURL)
```

Comment:
trying to think of ways to avoid "knowing" the path in order to test. One idea maybe a custom event that when "encoded" serializes bad data. Did not validate this idea though, but something worth to be done for the same reason as mentioned in the few tests above

Reply:
good idea, I'll incorporate in later PR